transport/src/main/java/io/scalecube/transport/Transport.java

```diff
@@ -335,6 +360,40 @@ public void operationComplete(ChannelFuture channelFuture) throws Exception {
     }
   }
 
+  private static InetAddress getNetworkInterfaceAddress(String intfc, boolean preferIPv6) {
```

Comment:
I propose to move logic from this method and from bind0 method to separate class (e.g. IpAddressChooser).